test/jdk/javax/swing/JSlider/8262981/JSliderAccessibleAction.java

```diff
@@ -0,0 +1,312 @@
+            invalidIncrementCountDownLatch.countDown();
+            updateJSliderValue(accessibleAction, jSliderValueLbl, INVALID_INCREMENT);
+            actionDescriptionValue.getAndSet(accessibleAction.getAccessibleActionDescription(INVALID_INCREMENT) == null);
+        });
+
+        decrementBtn = new JButton("Decrement");
+        decrementBtn.addActionListener((actionEvent) -> {
+            validDecrementCountDownLatch.countDown();
+            updateJSliderValue(accessibleAction, jSliderValueLbl, VALID_DECREMENT);
+            actionDescriptionValue.getAndSet(accessibleAction.getAccessibleActionDescription(VALID_DECREMENT).equals(AccessibleAction.DECREMENT));
+        });
+
+        incrementBtn = new JButton("Increment");
+        incrementBtn.addActionListener((actionEvent) -> {
+            validIncrementCountDownLatch.countDown();
+            updateJSliderValue(accessibleAction, jSliderValueLbl, VALID_INCREMENT);
+            actionDescriptionValue.getAndSet(accessibleAction.getAccessibleActionDescription(VALID_INCREMENT).equals(AccessibleAction.INCREMENT));
+        });
+
+        JPanel buttonPanel = new JPanel(new GridLayout(4, 1));
+        buttonPanel.add(invalidDecrementBtn);
+        buttonPanel.add(invalidIncrementBtn);
+        buttonPanel.add(decrementBtn);
+        buttonPanel.add(incrementBtn);
+        container.add(buttonPanel, BorderLayout.SOUTH);
+
+        jFrame.pack();
+        jFrame.setLocationRelativeTo(null);
+        jFrame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
+        jFrame.setVisible(true);
+    }
+
+    private void updateJSliderValue(AccessibleAction accessibleAction,
+                                    JLabel lbl, int value) {
+        accessibleAction.doAccessibleAction(value);
+        currentJSliderValue.getAndSet(jSlider.getValue());
+        lbl.setText("JSlider value : " + currentJSliderValue.get() + "%");
+    }
+
+    private static boolean setLookAndFeel(String lafName) {
+        try {
+            UIManager.setLookAndFeel(lafName);
+        } catch (UnsupportedLookAndFeelException unsupportedLookAndFeelException) {
+            System.out.println("Ignoring Unsupported laf : " + lafName);
+            return false;
+        } catch (ClassNotFoundException | InstantiationException
+                | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
+        return true;
+    }
+
+    public void test() throws AWTException,
+            InterruptedException, InvocationTargetException {
+        Robot robot = new Robot();
+        robot.setAutoDelay(300);
+        robot.waitForIdle();
+
+        List<String> instLookAndFeels =
+                Arrays.stream(UIManager.getInstalledLookAndFeels())
+                        .map(UIManager.LookAndFeelInfo::getClassName).collect(toList());
+
+        for (String laf : instLookAndFeels) {
+            try {
+                invalidDecrementCountDownLatch = new CountDownLatch(1);
```

Comment:
these fields are accessed both from EDT and the main test thread

Reply:
Fixed